cranelift/codegen/src/ir/dfg.rs

```diff
@@ -931,19 +930,26 @@ impl DataFlowGraph {
     where
         I: Iterator<Item = Option<Value>>,
     {
-        self.results[inst].clear(&mut self.value_lists);
+        self.clear_results(inst);
 
         let mut reuse = reuse.fuse();
         let result_tys: SmallVec<[_; 16]> = self.inst_result_types(inst, ctrl_typevar).collect();
         let num_results = result_tys.len();
+        debug_assert!(u16::try_from(num_results).is_ok(), "Too many result values");
 
-        for ty in result_tys {
-            if let Some(Some(v)) = reuse.next() {
+        for (expected, ty) in result_tys.into_iter().enumerate() {
+            let num = expected as u16;
```

Comment:
(I realize this is inlined code, but thoughts still follow:)

The `as u16` cast here is indeed protected by the assert above, but it's still something that catches my eye and takes a bit to understand safety thereof -- could we instead do the `try_from` here (overhead is the same in the common case of one result)?

Also, it occurs to me that we want this check even in release builds (if other limits somehow let through a many-result function, we don't want to rely on fuzzing catching a wraparound-caused miscompile before users do); so either a `try_from().expect()` or a full `assert!` is probably what we want.